projects/simple/src/Example/Project.hs

```diff
@@ -1,19 +1,49 @@
-module Example.Project (topEntity, plus) where
+-- @createDomain@ below generates a warning about orphan instances, but we like
+-- our code to be warning-free.
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Example.Project where
 
 import Clash.Prelude
 
--- | Add two numbers. Example:
+-- Create a domain with the frequency of your input clock. For this example we used
+-- 50 MHz.
+createDomain vSystem{vName="MyDomain", vPeriod=hzToPeriod 50e6}
+
+-- | 'topEntity' is Clash's equivalent of 'main' in other programming languages.
+-- Clash will look for it when compiling 'Example.Project' and translate it to
+-- HDL. While polymorphism can be used freely in Clash projects, a 'topEntity'
+-- must be monomorphic and must use non- recursive types. Or, to put it
+-- hand-wavily, a 'topEntity' must be translatable to a static number of wires.
 --
--- >>> plus 3 5
--- 8
-plus :: Signed 8 -> Signed 8 -> Signed 8
-plus a b = a + b
+-- The inputs need to be declared as parameters in order to be able to give them
+-- custom names. In other words, @topEntity = exposeClockResetEnable acc@ would
+-- prevent naming the inputs.
+topEntity ::
+  Clock MyDomain ->
+  Reset MyDomain ->
+  Enable MyDomain ->
+  Signal MyDomain (Unsigned 8) ->
+  Signal MyDomain (Unsigned 8)
+topEntity clk rst en inp = (exposeClockResetEnable accum) clk rst en inp
+-- Give names to the top entity parts
+{-# ANN topEntity
```

Comment:
It's interesting we're not annotating it `NOINLINE`. I think we should recommend people use Clash 1.8 with GHC 9.4 or newer, and that they should annotate `topEntity` `OPAQUE`. Add something to the `.cabal`  file that will prevent people from using it with GHC 9.2 or older, and write instructions that if they _do_ want to use an older GHC, they will need to replace `OPAQUE` annotations with `NOINLINE`.

I'm not sure if `NOINLINE` has an effect on a function that is not used in another function (you'd have to ask Martijn), but `OPAQUE` definitely does have an effect and should be added.

Maybe we need to update this in two places at a time, though: both in `Clash.Tutorial` and other starters docs we have and here.